domain/schema_validity.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package domain
+
+import (
+	"sync"
+	"time"
+
+	"github.com/pingcap/tidb/store/tikv/oracle"
+)
+
+// SchemaValidityInfo is the interface for checking the validity of schema version.
+type SchemaValidityInfo interface {
+	// Update the schema validity info, add a new item, delete the expired items.
+	// The schemaVer is valid within leaseGrantTime plus lease duration.
+	Update(leaseGrantTime uint64, schemaVer int64)
+	// Check is it valid for a transaction to use schemaVer, at timestamp txnTS.
+	Check(txnTS uint64, schemaVer int64) bool
+	// Latest returns the latest schema version it knows, but not necessary a valid one.
+	Latest() int64
+}
+
+type schemaValidityInfo1 struct {
```

Comment:
because we already have schemaValidityInfo, while this PR is a refact and schemaValidityInfo1 will replace the old schemaValidityInfo eventually.